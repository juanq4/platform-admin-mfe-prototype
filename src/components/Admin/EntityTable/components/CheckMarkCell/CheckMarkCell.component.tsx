import React, { memo, useMemo } from "react";
import { EntityTableClassName } from "../../EntityTable.definition";
import { CheckMarkCellIdModel } from "./CheckMarkCell.definition";
import type { CheckMarkCellProps } from "./CheckMarkCell.definition";

const CheckMarkCellBase = (props: CheckMarkCellProps): JSX.Element => {
  const { id, value } = props;

  const idModel = useMemo(() => new CheckMarkCellIdModel(id), [id]);

  if (!value) return <span id={idModel.id} />;

  return (
    <div
      className={`${EntityTableClassName.TableCell} ${EntityTableClassName.TableCellWithCenteredModifier} ${EntityTableClassName.TableCellWithIconModifier}`}
    >
      <span id={idModel.id} className="q4i-checkmark-4pt" />
    </div>
  );
};

export const CheckMarkCell = memo(CheckMarkCellBase);
