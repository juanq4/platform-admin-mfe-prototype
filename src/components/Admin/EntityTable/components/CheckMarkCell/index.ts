export * from "./CheckMarkCell.component";
export * from "./CheckMarkCell.definition";
export * from "./CheckMarkCell.utils";
