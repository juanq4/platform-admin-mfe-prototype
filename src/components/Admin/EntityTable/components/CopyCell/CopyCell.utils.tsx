import { CopyCell } from "./CopyCell.component";
import type { CopyCellListId, CopyCellProps } from "./CopyCell.definition";

export function mapIdToCopyCell(params: CopyCellProps, listId: CopyCellListId): JSX.Element {
  const id = listId?.getId(params?.data?.id);

  return <CopyCell {...params} id={id} />;
}
