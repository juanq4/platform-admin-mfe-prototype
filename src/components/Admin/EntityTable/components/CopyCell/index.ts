export * from "./CopyCell.component";
export * from "./CopyCell.definition";
export * from "./CopyCell.utils";
