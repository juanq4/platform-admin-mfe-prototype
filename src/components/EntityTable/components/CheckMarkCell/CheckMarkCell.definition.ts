import { AutomationListId, IdModelBase } from "@q4/nimbus-ui";
import type { Key } from "react";
import type { EntityTableCellPropsBase } from "../../EntityTable.definition";

export interface CheckMarkCellProps extends EntityTableCellPropsBase {
  value: boolean;
}

export class CheckMarkCellListId extends AutomationListId<CheckMarkCellIdModel> {
  constructor(parentId: string, prefix = "CheckMarkCell-", postfix?: string) {
    super(parentId, prefix, CheckMarkCellIdModel, postfix);
  }
}

export class CheckMarkCellIdModel extends IdModelBase {
  constructor(id: IdModelBase["id"], index?: Key, postfix?: string) {
    super(id, index, postfix);
  }
}
