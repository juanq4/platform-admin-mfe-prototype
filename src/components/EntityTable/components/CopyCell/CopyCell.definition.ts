import { AutomationListId, IdModelBase, isNullOrWhiteSpace } from "@q4/nimbus-ui";
import type { Key } from "react";
import { CopyButtonIdModel } from "../../../CopyButton/CopyButton.definition";
import type { EntityTableCellPropsBase } from "../../EntityTable.definition";

export interface CopyCellProps extends EntityTableCellPropsBase {
  value: string;
}

export class CopyCellListId extends AutomationListId<CopyCellIdModel> {
  constructor(parentId: string, prefix = "CopyCell-", postfix?: string) {
    super(parentId, prefix, CopyCellIdModel, postfix);
  }
}

export class CopyCellIdModel extends IdModelBase {
  value: string;
  copyButton: CopyButtonIdModel;

  constructor(id: IdModelBase["id"], index?: Key, postfix?: string) {
    super(id, index, postfix);

    if (isNullOrWhiteSpace(this.id)) {
      this.copyButton = new CopyButtonIdModel(null);
      return;
    }

    this.value = `${this.id}Value`;
    this.copyButton = new CopyButtonIdModel(`${this.id}CopyButton`);
  }
}
