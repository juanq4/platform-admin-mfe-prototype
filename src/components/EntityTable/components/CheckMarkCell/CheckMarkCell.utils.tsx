import { CheckMarkCell } from "./CheckMarkCell.component";
import type { CheckMarkCellListId, CheckMarkCellProps } from "./CheckMarkCell.definition";

export function mapIdToCheckMarkCell(params: CheckMarkCellProps, listId: CheckMarkCellListId): JSX.Element {
  const id = listId?.getId(params?.data?.id);

  return <CheckMarkCell {...params} id={id} />;
}
