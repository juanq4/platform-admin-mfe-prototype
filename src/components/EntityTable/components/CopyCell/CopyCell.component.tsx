import { isNullOrWhiteSpace } from "@q4/nimbus-ui";
import React, { memo, useMemo } from "react";
import { CopyButton } from "../../../CopyButton/CopyButton.component";
import { EntityTableClassName } from "../../EntityTable.definition";
import { CopyCellIdModel } from "./CopyCell.definition";
import type { CopyCellProps } from "./CopyCell.definition";

const CopyCellBase = (props: CopyCellProps): JSX.Element => {
  const { id, value } = props;

  const idModel = useMemo(() => new CopyCellIdModel(id), [id]);

  const content = useMemo(() => {
    if (isNullOrWhiteSpace(value)) return <span id={idModel.value}>-</span>;

    return (
      <>
        <span id={idModel.value}>{value}</span>
        <CopyButton id={idModel.copyButton.id} label="Org Id" value={value} />
      </>
    );
  }, [idModel, value]);

  return (
    <div id={idModel.id} className={EntityTableClassName.TableCell}>
      {content}
    </div>
  );
};

export const CopyCell = memo(CopyCellBase);
