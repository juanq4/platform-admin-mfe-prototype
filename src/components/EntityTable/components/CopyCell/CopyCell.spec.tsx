import React from "react";
import { NimbusConfig } from "../../../../__mocks__/contexts/NimbusConfig.mock";
import { getAppWrapper } from "../../../../__mocks__/utils/wrappers";
import { render, screen } from "../../../../utils/testUtils";
import { CheckMarkCellIdModel } from "../CheckMarkCell/CheckMarkCell.definition";
import { CopyCell } from "./CopyCell.component";
import { CopyCellListId } from "./CopyCell.definition";
import { mapIdToCopyCell } from "./CopyCell.utils";

const mockData = { id: "4dec165f-4093-4182-bd72-88754" };
const mockCopyCellValues = [null, "", "     ", mockData.id];

function testCopyCell(id: string): void {
  const element = screen.queryByTestId(id);
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
}

function renderWithConfigs(element: JSX.Element) {
  return <NimbusConfig.ConfigProvider>{element}</NimbusConfig.ConfigProvider>;
}

describe("CopyCell Component", () => {
  const idModel = new CheckMarkCellIdModel("MockCheckMarkCellId");

  test("7386010: [Given] no props are provided [Expect] the component should render", () => {
    render(renderWithConfigs(<CopyCell value={null} />), { container: getAppWrapper() });
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("7434643: [Given] value is 'null' [Expect] the checkmark to not be visible", () => {
    render(renderWithConfigs(<CopyCell id={idModel.id} value={mockCopyCellValues[0]} />));
    testCopyCell(idModel.id);
  });

  test("7434644: [Given] value is '' [Expect] the checkmark to not be visible", () => {
    render(renderWithConfigs(<CopyCell id={idModel.id} value={mockCopyCellValues[1]} />));
    testCopyCell(idModel.id);
  });

  test("7434645: [Given] value is '     ' [Expect] the checkmark to be visible", () => {
    render(renderWithConfigs(<CopyCell id={idModel.id} value={mockCopyCellValues[2]} />));
    testCopyCell(idModel.id);
  });

  test("7434646: Given] value is '4dec165f-4093-4182-bd72-88754' [Expect] the checkmark to be visible", () => {
    render(renderWithConfigs(<CopyCell id={idModel.id} value={mockCopyCellValues[3]} />));
    testCopyCell(idModel.id);
  });
});

describe("mapIdToCopyCell", () => {
  const listId = new CopyCellListId("MockCheckMarkCell");

  test("7386011: [Given] the function is passed not no props [Expect] the component should render", () => {
    render(renderWithConfigs(mapIdToCopyCell(null, null)), { container: getAppWrapper() });
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("7434647: [Given] value is 'null' [Expect] the checkmark to not be visible", () => {
    render(renderWithConfigs(mapIdToCopyCell({ data: { ...mockData }, value: mockCopyCellValues[0] }, listId)));
    const id = listId.getId(mockData.id);
    testCopyCell(id);
  });

  test("7434648: [Given] value is '' [Expect] the checkmark to not be visible", () => {
    render(renderWithConfigs(mapIdToCopyCell({ data: { ...mockData }, value: mockCopyCellValues[1] }, listId)));
    const id = listId.getId(mockData.id);
    testCopyCell(id);
  });

  test("7434649: [Given] value is '     ' [Expect] the checkmark to be visible", () => {
    render(renderWithConfigs(mapIdToCopyCell({ data: { ...mockData }, value: mockCopyCellValues[2] }, listId)));
    const id = listId.getId(mockData.id);
    testCopyCell(id);
  });

  test("7434650: [Given] value is '4dec165f-4093-4182-bd72-88754' [Expect] the checkmark to be visible", () => {
    render(renderWithConfigs(mapIdToCopyCell({ data: { ...mockData }, value: mockCopyCellValues[3] }, listId)));
    const id = listId.getId(mockData.id);
    testCopyCell(id);
  });
});
