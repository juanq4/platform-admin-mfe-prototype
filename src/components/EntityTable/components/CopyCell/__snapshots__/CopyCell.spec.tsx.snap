// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyCell Component 7386010: [Given] no props are provided [Expect] the component should render 1`] = `
<div
  id="root"
>
  <div
    class="entity-table_table-cell"
  >
    <span>
      -
    </span>
  </div>
</div>
`;

exports[`CopyCell Component 7434643: [Given] value is 'null' [Expect] the checkmark to not be visible 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellId"
>
  <span
    id="MockCheckMarkCellIdValue"
  >
    -
  </span>
</div>
`;

exports[`CopyCell Component 7434644: [Given] value is '' [Expect] the checkmark to not be visible 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellId"
>
  <span
    id="MockCheckMarkCellIdValue"
  >
    -
  </span>
</div>
`;

exports[`CopyCell Component 7434645: [Given] value is '     ' [Expect] the checkmark to be visible 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellId"
>
  <span
    id="MockCheckMarkCellIdValue"
  >
    -
  </span>
</div>
`;

exports[`CopyCell Component 7434646: Given] value is '4dec165f-4093-4182-bd72-88754' [Expect] the checkmark to be visible 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  background-color: #0f5ca3;
  color: #ffffff;
  border-color: transparent;
  font-family: inherit;
  height: 40px;
  padding: 12px 16px;
  border-width: 1px;
  border-style: solid;
  font-size: 14px;
  line-height: 16px;
  text-transform: none;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  cursor: pointer!important;
  border-radius: 4px;
  padding: 0;
  width: 40px;
  text-align: center;
}

.emotion-0:hover {
  background: #0a3d6d;
  color: #ffffff;
}

.emotion-0:hover [data-type="nui-tooltip"] {
  opacity: 1;
  visibility: visible;
}

.emotion-0:focus {
  background: #0a3d6d;
  border-color: #4ea1ef;
  box-shadow: 0px 0px 3px #4ea1ef;
}

.emotion-0:active {
  background: #051f36;
}

.emotion-0:disabled {
  pointer-events: none;
  cursor: default;
  background: #f7fafc;
  color: #61707c;
  border-color: transparent;
}

@media only screen and (max-width: 768px) {
  .emotion-0 {
    height: 40px;
    padding: 12px;
    width: 40px;
  }
}

.emotion-1 {
  font-size: 12px;
  line-height: 40px;
}

.emotion-1+span {
  margin-left: 4px;
}

@media only screen and (max-width: 768px) {
  .emotion-1 {
    line-height: 16px;
  }
}

<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellId"
>
  <span
    id="MockCheckMarkCellIdValue"
  >
    4dec165f-4093-4182-bd72-88754
  </span>
  <button
    class="copy-button emotion-0"
    id="MockCheckMarkCellIdCopyButton"
    type="button"
  >
    <i
      class="copy-button_icon q4i-clone-duplicate-4pt emotion-1"
      id="MockCheckMarkCellIdCopyButtonIcon"
    />
  </button>
</div>
`;

exports[`mapIdToCopyCell 7386011: [Given] the function is passed not no props [Expect] the component should render 1`] = `
<div
  id="root"
>
  <div
    class="entity-table_table-cell"
  >
    <span>
      -
    </span>
  </div>
</div>
`;

exports[`mapIdToCopyCell 7434647: [Given] value is 'null' [Expect] the checkmark to not be visible 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754"
>
  <span
    id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754Value"
  >
    -
  </span>
</div>
`;

exports[`mapIdToCopyCell 7434648: [Given] value is '' [Expect] the checkmark to not be visible 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754"
>
  <span
    id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754Value"
  >
    -
  </span>
</div>
`;

exports[`mapIdToCopyCell 7434649: [Given] value is '     ' [Expect] the checkmark to be visible 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754"
>
  <span
    id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754Value"
  >
    -
  </span>
</div>
`;

exports[`mapIdToCopyCell 7434650: [Given] value is '4dec165f-4093-4182-bd72-88754' [Expect] the checkmark to be visible 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  background-color: #0f5ca3;
  color: #ffffff;
  border-color: transparent;
  font-family: inherit;
  height: 40px;
  padding: 12px 16px;
  border-width: 1px;
  border-style: solid;
  font-size: 14px;
  line-height: 16px;
  text-transform: none;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  cursor: pointer!important;
  border-radius: 4px;
  padding: 0;
  width: 40px;
  text-align: center;
}

.emotion-0:hover {
  background: #0a3d6d;
  color: #ffffff;
}

.emotion-0:hover [data-type="nui-tooltip"] {
  opacity: 1;
  visibility: visible;
}

.emotion-0:focus {
  background: #0a3d6d;
  border-color: #4ea1ef;
  box-shadow: 0px 0px 3px #4ea1ef;
}

.emotion-0:active {
  background: #051f36;
}

.emotion-0:disabled {
  pointer-events: none;
  cursor: default;
  background: #f7fafc;
  color: #61707c;
  border-color: transparent;
}

@media only screen and (max-width: 768px) {
  .emotion-0 {
    height: 40px;
    padding: 12px;
    width: 40px;
  }
}

.emotion-1 {
  font-size: 12px;
  line-height: 40px;
}

.emotion-1+span {
  margin-left: 4px;
}

@media only screen and (max-width: 768px) {
  .emotion-1 {
    line-height: 16px;
  }
}

<div
  class="entity-table_table-cell"
  id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754"
>
  <span
    id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754Value"
  >
    4dec165f-4093-4182-bd72-88754
  </span>
  <button
    class="copy-button emotion-0"
    id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754CopyButton"
    type="button"
  >
    <i
      class="copy-button_icon q4i-clone-duplicate-4pt emotion-1"
      id="MockCheckMarkCellCopyCell-4dec165f-4093-4182-bd72-88754CopyButtonIcon"
    />
  </button>
</div>
`;
